dateparser/date.py

```diff
@@ -39,6 +39,7 @@
 RE_NBSP = re.compile(u'\xa0', flags=re.UNICODE)
 RE_SPACES = re.compile(r'\s+')
 RE_TRIM_SPACES = re.compile(r'^\s+(\S.*?)\s+$')
+RE_TRIM_COLONS = re.compile(r':*(\S.*?):*$')
```

Comment:
I don’t understand how it works. I would expect that greedy `:*` at the beginning to cause issues, but it seems to behave as non-greedy here, as opposed to `:+`:

```
>>> re.sub(r':+(\S.*?):*$', r'\1', '2019-07-31 12:00:')
'2019-07-31 1200'
>>> re.sub(r':*(\S.*?):*$', r'\1', '2019-07-31 12:00:')
'2019-07-31 12:00'
```

In any case, what would you think about adding a `^` at the beginning of the expression to make it more obvious that the `:` capture before the capture group is only meant to strip colons from the beginning?

Reply:
I think we can safely change to `^`. I added at the begining to be more safe, but I think we don't need it.